frontend/src/components/associations/settings/AssociationOptInSettings.tsx

```diff
@@ -69,15 +69,14 @@ const EnableModuleForm = ({
   });
 
   return (
-    <label
-      onClick={() => {
-        save({ id: id, enabled: !checked });
-      }}
-    >
+    <label>
       <input
         type="checkbox"
         className="custom-switch-input"
         checked={checked}
+        onChange={() => {
+          save({ id: id, enabled: !checked, association: id, products: [] });
```

Comment:
This will reset the list of product each time right? This may be dangerous in case of misclick

Reply:
The products field will only be used if the marketplace doesn't exist yet, in which case it will be created with an empty products list. In case of an update, the 'products' value will not be used as the update method from the MarketplaceViewSet only changes the 'enabled' boolean.

Reply:
So if we always initialize the product list to an empty list, why do we specify it at all?